### PR TITLE
Protect: Add user connection gate for applicable actions

### PR DESCRIPTION
**Pull request by dkmyta**
#### Description
With the introduction of the paid UI as part of the Scan in Protect integration, actions are now available to the users (fix and ignore threats) that will require a user connection. This PR establishes a user connection gate for these particular actions.

#### Changes proposed in this Pull Request:

* Adds a `user-connection-gate` component that checks for the presence of a user connection and responds accordingly
* Adds a `user-connection-needed-modal` component that displays in the absence of a user connection only when the process of fixing or ignoring a threat is initiated
* Clarifies what actions require a connection, why we need it, and how to go about connecting
* Redirects to the Protect admin page and presents the appropriate modal after the connection is established

#### Other information:
- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

- 1201069996155224-as-1203180130348952

#### Does this pull request change what data or activity we track or use?

- No

#### Testing instructions:

* Start up a Jurassic Ninja instance w/ the Jetpack Beta plugin, a Scan level plan, and this Protect branch installed
* Install a known threat
* Activate Protect, select the Free connection (site-only) and allow the initial report to generate
* Run JPDB to ensure your connection is site-only, and downgrade if necessary
* Return to Protect admin page, select Fix all threats (#), Ignore threat, and Fix threat and verify that for each you are presented with a prompt for a user connection
* Proceed to connect your user and be redirected back to the Protect admin page
* Verify that each of the above buttons now present the following:
   * Fix all threats (#) - a modal prompt for server credentials
   * Fix threat - a modal prompt for server credentials
   * Ignore threat - a modal prompt to confirm or cancel the ignore threat action
* Ensure there are no regressions in functionality

#### Screenshots
Screen Shot 2022-10-18 at 07 05 18

### CODE REVIEW

File: projects/plugins/protect/src/js/components/credentials-needed-modal/index.jsx [hunk 1]
```diff
@@ -59,7 +59,7 @@ const CredentialsNeededModal = () => {
 
 			<Text mb={ 3 }>
 				{ __(
-					'Once you’ve entered server credentials, Jetpack will be fixing the selected threats.',
+					'Once you’ve entered server credentials, Jetpack Protect will be fixing the three selected threats.',
```
Comment (nateweller) [inline projects/plugins/protect/src/js/components/credentials-needed-modal/index.jsx]:
```suggestion
					'Once you’ve entered server credentials, Jetpack Protect will be fixing the selected threats.',
```
Minor regression here with the "three" being reintroduced, probably left over from resolving a merge conflict or something like that.

File: projects/plugins/protect/src/js/components/ignore-threat-modal/styles.module.scss [hunk 2]
```diff
@@ -38,3 +38,21 @@
     display: flex;
     justify-content: space-between;
 }
+
```
Comment (nateweller) [inline projects/plugins/protect/src/js/components/ignore-threat-modal/styles.module.scss]: Looks like this CSS is unused and can probably be removed :+1:

**Comment by nateweller** [comment]
Works perfectly! Nice consistent approach with the gate component pattern.

I think the copy will work as it is, but I was thinking that we could run the need for this new modal by the design team to see if there is any input about how we describe the need for the user connection. The concept of a "user connection" is familiar to us, but I think the average user might just ask "user connection _to what_?". Maybe we could reach out to Eder for input? I think we're still good to merge this once [this](https://github.com/Automattic/jetpack/pull/26909/files#r1001109289) is committed.

**Comment by dkmyta** [comment]
>I think the copy will work as it is, but I was thinking that we could run the need for this new modal by the design team to see if there is any input about how we describe the need for the user connection. The concept of a "user connection" is familiar to us, but I think the average user might just ask "user connection to what?". Maybe we could reach out to Eder for input?

I agree with this - I kind of threw it together following the same pattern as the server credential modal, but I wasn't overly satisfied with my descriptions and feel that some of the content may just be a little redundant.

**Review by nateweller — APPROVED** [review]
LGTM!

**Pull request merged**